Improve label explanation

lontivero opened the pull request:
This PR implements a suggestion found in this article https://uxdesign.cc/the-mythical-privacy-in-the-age-of-bitcoin-and-what-were-doing-about-it-bd9d166d9dfa

![image](https://user-images.githubusercontent.com/127973/64176166-33d8a200-ce33-11e9-8d3e-9a6fd4e79b5f.png)

nopara73 commented:
Concept NACK. Way too chatty.

lontivero commented:
What about the watermark "What is this address for?"?

lontivero commented:
Is this better? It is simply the tool tip.

![image](https://user-images.githubusercontent.com/127973/64185690-f0d2fa80-ce43-11e9-9b26-ea4e8ba7e5b3.png)

nopara73 commented:
I can't spot the difference.

benthecarman commented on WalletWasabi.Gui/Icons/Icons.xaml:
```diff
@@ -142,7 +147,7 @@
       <DrawingGroup x:Key="LoadWallet_Folder">
         <DrawingGroup.Children>
-          <GeometryDrawing Brush="#FFF6F6F6" Geometry="F1 M 19,50L 28,34L 63,34L 54,50L 19,50 Z M 19,28.0001L 35,28C 36,25 37.4999,24.0001 37.4999,24.0001L 48.75,24C 49.3023,24 50,24.6977 50,25.25L 50,28L 53.9999,28.0001L 53.9999,32L 27,32L 19,46.4L 19,28.0001 Z " />
+          <GeometryDrawing Brush="#FFB3B3B3" Geometry="F1 M 19,50L 28,34L 63,34L 54,50L 19,50 Z M 19,28.0001L 35,28C 36,25 37.4999,24.0001 37.4999,24.0001L 48.75,24C 49.3023,24 50,24.6977 50,25.25L 50,28L 53.9999,28.0001L 53.9999,32L 27,32L 19,46.4L 19,28.0001 Z " />
```
Is this supposed to be here?

lontivero replied:
No, it should not. Thanks for reviewing.

yahiheb commented:
It is important for users to understand that.
How about these sentences:
`Labels help identify different coins received to not get merged.`
`Labels help identify coins received to not get merged.`
I think we shouldn't use the word mixed here.

nopara73 commented:
Yes, we shouldn't use the word mixed, nor the word coins, nor the word merged. Receive tab is the very first thing the user encounters with and you'll induce a WTF moment right away. The current text is good. @mayankchhabra I think you sparked this PR with your article, so please chime in.

mayankchhabra commented:
IMO the current receive tab in itself, with no explanation of what a label is, induces a WTF moment right away. 

For a user who has never used Wasabi (or a wallet that implements labels) labeling is a foreign concept. 

Chatty UI helps ease the UX by letting go of negative emotions such as fear and uncertainty. 

I’d say “Labels help you easily identify your Bitcoin addresses. Eg. ‘August Salary” is a short and sweet version that does a decent job, and should be placed under the input field so it’s visible at all times without the need of hovering.

yahiheb commented:
> nor the word coins, nor the word merged.

If `Watermark="Label (Payment reason)"`

Maybe one of the following sentences will be good:
`Payment reasons help future spending by not combining coins.`
`Payment reasons guard privacy by spending coins separately.`

and it fits well with:
`"Example: "August Salary", "Investment return", etc."`

yahiheb commented on WalletWasabi.Gui/Controls/WalletExplorer/ReceiveTabView.xaml:
```diff
@@ -25,15 +25,21 @@
             <Grid ColumnDefinitions="*,300" DockPanel.Dock="Top" Margin="0 0 0 10">
               <StackPanel Spacing="8">
-                <controls:ExtendedTextBox Text="{Binding Label, Mode=TwoWay}" Watermark="Label (Required)" CaretIndex="{Binding CaretIndex}" UseFloatingWatermark="True" Grid.Column="0" Margin="0 0 10 0">
+                <controls:ExtendedTextBox Text="{Binding Label, Mode=TwoWay}" Watermark="Label (What is this address for?)" CaretIndex="{Binding CaretIndex}" UseFloatingWatermark="True" Grid.Column="0" Margin="0 0 10 0">
```
```suggestion
                <controls:ExtendedTextBox Text="{Binding Label, Mode=TwoWay}" Watermark="Label (Payment reason)" CaretIndex="{Binding CaretIndex}" UseFloatingWatermark="True" Grid.Column="0" Margin="0 0 10 0">
```
See [comment](https://github.com/zkSNACKs/WalletWasabi/pull/2205#issuecomment-527936296)

lontivero commented:
I agree with @mayankchhabra here, labeling is an new concept for many users, they don't what labeling is for.
About this PR, IMO we should not force users to read long texts by default but in case you need help the tool tip should contain some explanation and/or example.  @yahiheb's suggestions look good too.

mayankchhabra commented:
Can we substitute the word "payment"? It naturally implies the sense of giving (or sending) and not receiving, and might not come off as an **inward** payment at first take...

nopara73 commented:
Still concept NACK for the same reasons. Also note the label is not a payment reason, it's a "Comma separated string on the entities and people who will know the link between the wallet's user the generated address and the transactions arriving to it."

So, if Alice sends me 3BTC from MtGox, then the proper label would be "Alice, MtGox".

yahiheb commented:
> Also note the label is not a payment reason, it's a "Comma separated string ....
So, if Alice sends me 3BTC from MtGox, then the proper label would be "Alice, MtGox".

I completely agree, however the problem is the users won't know this with the tip saying:
`Start labeling today and your privacy will thank you tomorrow!`

How about `Payment info` or `Transaction info`
`Transaction info help future spending by not combining coins.`
`Transaction info guard privacy by spending coins separately.`

lontivero commented:
@nopara73 I think we all agree with what you say. What we are trying to do here is to find a better way to explain that to the users, that's why imho the concept is okay and we should discuss how to achieve it. For what I read in this thread it seems there is some level of consensus about the need to improve this somehow.

molnard commented:
Better options for `Payment reason`:

1. Transaction Notes
2. Transaction Comment
3. Transaction Reason
4. Transaction Why?

My preference is `Transaction Notes`.

nopara73 commented:
Let me repeat myself with some accuracy improvements and please this time pay attention: **A label is a comma separated string that consists of entities and people, who are suspected or known to know the link between the wallet user and the generated address/made transaction/created transaction output, depending on which construct the label is applied to.**

nopara73 commented:
> entities and people, who are suspected or known to know the link

So if you buy candy in the candy shop, the label is the candy shop and maybe bitpay, not the candy itself. Thus all your suggestions are misguided.

lontivero commented:
That's exactly who I do it. Here you can see that I moved some btc from electrum (unknown source), then I mixed it and sent a fraction to Pablo. 

![image](https://user-images.githubusercontent.com/127973/65242290-0428cb80-dabc-11e9-82d6-837ed9d60cf7.png)

I bet my hair that most users don't do it in this way because Wasabi doesn't give them a tip/clue. So, what if in the tool tip we explain this as @nopara73 explained in this issue? It could say  `A label is a comma separated string that consists of entities and people, who are suspected or known to know the link between the wallet user and the generated address` (I removed the last part because it applies to the send label.

nopara73 commented:
> I bet my hair that most users don't do it in this way because Wasabi doesn't give them a tip/clue. 

Wasabi was actively misleading them with the "ZeroLink.... and change of..." labels. It's not the case anymore.

lontivero commented:
What about something like this? Is it better?

![image](https://user-images.githubusercontent.com/127973/65265611-b9bd4400-dae7-11e9-98a7-8463886e37f1.png)

yahiheb commented:
`Watermark="Any person/entity that knows this address is yours (Required)"`
`Text="If you buy a candy from a candy shop, the label is the candy shop and maybe BitPay (payment processor)."`

mayankchhabra commented:
> Who knows this address is yours (Required)

That's a good way to put it, BUT as a newbie (or not), a user might just end up putting "nobody" or "just me" as the label because nobody, except the user, knows the address or its ownership at the time of creation, until its shared or transacted with.

lontivero commented:
Ok, I got it but the text cannot be longer, in fact it should be shorter. If that text is not good then we should go back to `Label (Required)` and put the explanation on the tool tip.

yahiheb commented:
> > Who knows this address is yours (Required)
> 
> That's a good way to put it, BUT as a newbie (or not), a user might just end up putting "nobody" or "just me" as the label because nobody, except the user, knows the address or its ownership at the time of creation, until its shared or transacted with.

how about this:
`Any person/entity who would know that this address is yours`

MaxHillebrand commented:
Good conversation here!!

I agree that the text has to change, and I like `Any person/entity who would know that this address is yours` as this conveys the actual reason why labeling is important.

molnard commented:
I would go with the questioning format what @mayankchhabra suggested but refactored.
- The watermark in the textbox: 
Add a label(s) to remember: with who will you share this address?
- Tooltip
Any person/entity who knows this address will also know future transactions related to this address.
